src/kontrol/kompile.py

```diff
@@ -75,6 +77,12 @@ def foundry_kompile(
         if regen or not req_path.exists():
             _LOGGER.info(f'Copying requires path: {req} -> {req_path}')
             shutil.copy(req, req_path)
+            # If the copied file is not writeable
+            if not os.access(req_path, os.W_OK):
+                # Fetch current permissions
+                current_permissions = os.stat(req_path).st_mode
+                # Grant write permissions
+                os.chmod(req_path, current_permissions | stat.S_IWUSR | stat.S_IWGRP | stat.S_IWOTH)
```

Comment:
Why is it necessary to have write access on these requires?

Reply:
We want to be able to delete and replace them when `kontrol build --rekompile` is executed. The files not copied from `kontrol/kdist` had write access previously, but the we are not copying from `kdist` (such as `keccak.md`) don't, and it causes issues such as this one:
```
INFO 2024-08-23 08:58:19,927 kontrol.kompile - Copying requires path: /nix/store/g67pca4v4lx9zgna190kdwvrj4n0jarm-python3.10-kontrol-0.1.409/lib/python3.10/site-packages/kontrol/kdist/keccak.md ->                           
out/kompiled/requires/keccak.md                                                                                                                                                                                                
An error occurred while building your Kontrol project: [Errno 13] Permission denied: 'out/kompiled/requires/keccak.md'
```

Reply:
> it causes issues such as this one

Please include the issue in the PR description.

Reply:
Done, thank you!